python3/dnf_plugins/ptoken.py

```diff
@@ -0,0 +1,18 @@
+"""dnf plugin to add ptoken for repos"""
+import dnf
+
+class Ptoken(dnf.Plugin):
+    """ptoken plugin class"""
+    name = "ptoken"
+
+    def config(self):
+        with open('/etc/xensource/ptoken', encoding="utf-8") as file:
+            ptoken = file.read().strip()
+
+        for repo_name in self.base.repos:
+            repo = self.base.repos[repo_name]
+
+            if len(repo.baseurl) > 0 and repo.baseurl[0].startswith("http://127.0.0.1") \
+                and repo.ptoken:
```

Comment:
This is a newly added config item for dnf? May I please know the reason to add this?

Reply:
Would this require changes on xapi code which populate the repo configurations?

Reply:
do you mean `ptoken`?

No, this is an optional configure item for the repo, to decide whether we should enable pool token for a repo. 
it is used by yum, and I patched dnf to support this.

No extra xapi code change is required for this, on the other side,  dnf is pathed to support this as xapi use it.